analyzers/tests/SonarAnalyzer.UnitTest/SymbolicExecution/Roslyn/RoslynSymbolicExecutionTest.PatternMatching.cs

```diff
@@ -117,5 +118,77 @@ public void IsPattern_OtherPattern_VisitsBothBranches()
                 .And.ContainSingle(x => x.HasConstraint(BoolConstraint.True))
                 .And.ContainSingle(x => x.HasConstraint(BoolConstraint.False));
         }
+
+        [TestMethod]
+        public void RecursivePattern_NoSymbol_DoesNothing()
+        {
+            const string code = @"
+if (arg is { })
+{
+    Tag(""ArgNotNull"", arg);
+}
+Tag(""End"", arg);";
+            var validator = SETestContext.CreateCS(code, ", object arg").Validator;
+            validator.ValidateContainsOperation(OperationKind.RecursivePattern);
+            validator.ValidateTag("ArgNotNull", x => x.Should().BeNull());  // ToDo: Should learn that it is not null too
```

Comment:
// ToDo: Should learn that it is not null too
 
Then we should also add a test, to make sure that if `arg` has `NotNull` before, it has the constraint only set once.

Reply:
Same constraint cannot be set twice by-design. If something is set to `NotNull`, it will even throw away the `Null` one.